### PR TITLE
Improvement: Make ZLib dependency optional

### PR DESCRIPTION
**Pull request by oncaphillis**
* New CMake option WITH_DLT_FILETRANSFER
    * Only request ZLIB package if
      WITH_DLT_COREDUMPHANDLER==ON or
      WITH_DLT_FILETRANSFER==ON

Signed-off-by: Sebastian Kloska <sebastian.kloska@snafu.de>

### CODE REVIEW

**Comment by votrungchi** [comment]
Hi @oncaphillis dlt-system use zlib based compression dlt-system-filetransfer feature. So dlt-system also depends on the zlib library. Thanks.

**Comment by votrungchi** [comment]
@oncaphillis Looks good to me. Just some minor comments. Thanks for you work.

File: src/system/CMakeLists.txt [hunk 1]
```diff
@@ -18,20 +18,33 @@ if(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
...
 
 set(dlt_system_SRCS dlt-system.c dlt-system-options.c dlt-system-process-handling.c
-	dlt-system-filetransfer.c dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
+   dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
 	dlt-system-syslog.c dlt-system-watchdog.c dlt-system-journal.c)
+
+ if(WITH_DLT_FILETRANSFER)
+  set(dlt_system_SRCS ${dlt_system_SRCS} dlt-system-filetransfer.c)
+  add_definitions(-DDLT_FILETRANSFER_ENABLE)
+  message( STATUS "Building dlt-system WITH filetransfer support")
```
Comment (votrungchi) [inline src/system/CMakeLists.txt]: We should put the STATUS on the STATUS section in top CMakeLists.txt at line 239 (after WITH_DLT_SYSTEM), so we can remove this one and line 29 as well.

File: src/system/CMakeLists.txt [hunk 2]
```diff
@@ -18,20 +18,33 @@ if(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
 endif(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
 
 set(dlt_system_SRCS dlt-system.c dlt-system-options.c dlt-system-process-handling.c
-	dlt-system-filetransfer.c dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
+   dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
```
Comment (votrungchi) [inline src/system/CMakeLists.txt]: Please correct the indentation.

File: src/system/CMakeLists.txt [hunk 3]
```diff
@@ -18,20 +18,33 @@ if(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
...
 	dlt-system-syslog.c dlt-system-watchdog.c dlt-system-journal.c)
+
+ if(WITH_DLT_FILETRANSFER)
```
Comment (votrungchi) [inline src/system/CMakeLists.txt]: Please correct the indentation.

File: CMakeLists.txt [hunk 4]
```diff
@@ -63,6 +63,7 @@ option(WITH_DLT_SHM_ENABLE    "EXPERIMENTAL! Set to ON to use shared memory as I
 option(WITH_DLT_ADAPTOR       "Set to ON to build src/adaptor binaries"                                          OFF)
 option(WITH_DLT_CONSOLE       "Set to ON to build src/console binaries"                                          ON)
 option(WITH_DLT_EXAMPLES      "Set to ON to build src/examples binaries"                                         ON)
+option(WITH_DLT_FILETRANSFER  "Set to ON to build dlt-system with filetransfer support"                          ON)
```
Comment (ssugiura) [inline CMakeLists.txt]: I would like to set the default to OFF because WITH_DLT_SYSTEM is set to OFF and this only works on Linux.

Comment (oncaphillis) [reply CMakeLists.txt]: ACK

Comment (ssugiura) [reply CMakeLists.txt]: @oncaphillis Looks like you've changed the wrong line to OFF...?

Comment (oncaphillis) [reply CMakeLists.txt]: :-(

Comment (oncaphillis) [reply CMakeLists.txt]: ACK again

**Pull request merged**